### PR TITLE
fix: update props in fragments together with other fragments which ar…

### PR DESCRIPTION
**Pull request by nightnei**
…e replacing at a page

### CODE REVIEW

**Comment by b1ff** [comment]
Please add tests

**Review by b1ff — CHANGES_REQUESTED** [review]
Please add tests

File: ilc/client/ClientRouter.js [hunk 1]
```diff
@@ -212,29 +212,34 @@ export default class ClientRouter extends EventEmitter {
...
                     delete this.#activeApps.current[slotName];
                 }
             });
 
-            if (appsToForceRerender.length) {
-                window.addEventListener('single-spa:app-change', () => {
-                    this.#logger.log(`ILC: Triggering app re-mount for [${appsToForceRerender.map(n => n.appName)}] due to changed props.`);
+            window.addEventListener('single-spa:app-change', () => {
+                if (updateEventsToTrigger.length) {
+                    updateEventsToTrigger.forEach(this.emit.bind(this));
```
Comment (b1ff) [inline ilc/client/ClientRouter.js:234]: as far as I understand `triggerAppChange` emits `ilc:before:routing`, probably emmiting update events before `ilc:before:routing` is triggered wouldn't be correct

**Review by b1ff — CHANGES_REQUESTED** [review]

File: ilc/client/ClientRouter.js [hunk 2]
```diff
@@ -212,29 +212,34 @@ export default class ClientRouter extends EventEmitter {
...
-                    appsToForceRerender.push({ slotName, appName })
+                    appsToForceRerender.push(appName)
                     delete this.#activeApps.current[slotName];
                 }
             });
 
-            if (appsToForceRerender.length) {
-                window.addEventListener('single-spa:app-change', () => {
-                    this.#logger.log(`ILC: Triggering app re-mount for [${appsToForceRerender.map(n => n.appName)}] due to changed props.`);
+            window.addEventListener('single-spa:app-change', () => {
```
[the remark this answers is not in the document]
Comment (nightnei) [reply ilc/client/ClientRouter.js]: have done for all ilc and singleSpa related events

File: ilc/client/ClientRouter.js [hunk 3]
```diff
@@ -212,29 +212,34 @@ export default class ClientRouter extends EventEmitter {
         const appsWithDifferentProps = this.#getAppsWithDifferentProps(this.#prevRoute.slots, this.#currentRoute.slots);
         if (appsWithDifferentProps.length) {
             const appsToForceRerender = [];
+            const updateEventsToTrigger = [];
 
             appsWithDifferentProps.forEach(({ slotName, appName }) => {
                 const eventUpdateFragment = `ilc:update:${slotName}_${appName}`;
```
[the remark this answers is not in the document]
Comment (nightnei) [reply ilc/client/ClientRouter.js]: done

File: ilc/client/ClientRouter.js (hunk 1 above)
Comment (nightnei) [reply ilc/client/ClientRouter.js]: done

**Review by b1ff — APPROVED** [review]

**Pull request merged**